src/plugins/analysis/cve_lookup/code/cve_lookup.py

```diff
@@ -1,86 +1,66 @@
 from plugins.mime_blacklists import MIME_BLACKLIST_NON_EXECUTABLE
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from objects.file import FileObject
 
 try:
-    from ..internal.database_interface import QUERIES, DatabaseInterface
-    from ..internal.helper_functions import replace_characters_and_wildcards, unescape
+    from ..internal.lookup import Lookup
+    from ..internal.database.db_connection import DbConnection
 except ImportError:
     sys.path.append(str(Path(__file__).parent.parent / 'internal'))
-    from database_interface import QUERIES, DatabaseInterface
-    from helper_functions import replace_characters_and_wildcards, unescape
-
-MAX_TERM_SPREAD = (
-    3  # a range in which the product term is allowed to come after the vendor term for it not to be a false positive
-)
-MAX_LEVENSHTEIN_DISTANCE = 0
-DOTTED_VERSION_REGEX = re.compile(r'^[a-zA-Z0-9\-]+(\\\.[a-zA-Z0-9\-]+)+$')
-VALID_VERSION_REGEX = re.compile(r'v?(\d+!)?\d+(\.\d+)*([.-]?(a(lpha)?|b(eta)?|c|dev|post|pre(view)?|r|rc)?\d+)?')
-
-
-class Product(NamedTuple):
-    vendor_name: str
-    product_name: str
-    version_number: str
+    from lookup import Lookup
+    from database.db_connection import DbConnection
 
-
-class CveDbEntry(NamedTuple):
-    cve_id: str
-    vendor: str
-    product_name: str
-    version: str
-    cvss_v2_score: str
-    cvss_v3_score: str
-    version_start_including: str
-    version_start_excluding: str
-    version_end_including: str
-    version_end_excluding: str
+DB_PATH = str(Path(__file__).parent / '../internal/database/cve_cpe.db')
+KEYS_TO_IGNORE = ['summary', 'analysis_date', 'plugin_version', 'system_version']
 
 
 class AnalysisPlugin(AnalysisBasePlugin):
-    '''
+    """
     lookup vulnerabilities from CVE feeds using ID from CPE dictionary
-    '''
+    """
 
     NAME = 'cve_lookup'
     DESCRIPTION = 'lookup CVE vulnerabilities'
     MIME_BLACKLIST = MIME_BLACKLIST_NON_EXECUTABLE
     DEPENDENCIES = ['software_components']
-    VERSION = '0.0.5'
+    VERSION = '0.0.6'
     FILE = __file__
 
-    def process_object(self, file_object):
+    def process_object(self, file_object: FileObject) -> FileObject:
+        """
+        Process the given file object and look up vulnerabilities for each software component.
+        """
         cves = {'cve_results': {}}
-        for component in file_object.processed_analysis['software_components']['summary']:
-            product, version = self._split_component(component)
-            if product and version:
-                vulnerabilities = look_up_vulnerabilities(product_name=product, requested_version=version)
-                if vulnerabilities:
-                    cves['cve_results'][component] = vulnerabilities
+        connection = DbConnection(f'sqlite:///{DB_PATH}')
+        lookup = Lookup(connection)
+        for key, value in file_object.processed_analysis['software_components'].items():
+            if key not in KEYS_TO_IGNORE:
```

Comment:
since #1027 it should be `file_object.processed_analysis['software_components']['result'].items()` and filtering the keys with KEYS_TO_IGNORE should no longer be necessary if I'm not mistaken